### DISCUSSION TITLE
Typescript - Get model Payload from model name

### DISCUSSION THREAD

**Discussion by karimpanacci** [prisma/prisma#22504] [Q&A]
### Question

I wrote a typescript type that extrapolates the name of all the models from the PrismaClient instance.
so if PrismaClient is done like this:
```typescript
prisma = {
    user:{
        findUnique(){},
        findFirst(){}
    },
    notifications: "etc..."
}
```
then my type will be:
`type PrismaModelEnum = "user" | "notifications"`

I would like a way to get the payload that you would get using `GetUserPayload<typeof parameters>`, but taking the model dynamically based on `PrismaModelEnum`

**Comment by ludralph** [comment]
Hi @karim2012 👋 

Thank you for raising this question. 

I have shared this internally to see if this is possible. I'll get back to you with a response or another team member will respond.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
